operations/mimir/alertmanager.libsonnet

```diff
@@ -40,9 +40,10 @@
   alertmanager_pvc::
     if $._config.alertmanager_enabled then
       pvc.new() +
+      pvc.mixin.spec.withStorageClassName($._config.alertmanager_data_disk_class) +
```

Comment:
Would this work properly with a `null` value? It would be great to ensure that this can be applied to existing statefulsets.

Reply:
Yes, `null` does work, If `null` is defined, the storageClass will be omitted. Actually, it would probably make sense to define `null` ad default value in `config.libsonnet` to avoid problems during update since we can't update the `template` spec of an existing sts

Reply:
`null` does not work as you'd expect in this occasion as `withStorageClassName` is a function and the value itself might not necessarily end up as part of the key. 

You use the value to conditionally not include the whole `pvc.mixin.spec.withStorageClassName($._config.alertmanager_data_disk_class)` like we do with `hasFallbackConfig` as an example, but as is this does not work as intended.

![Screenshot 2023-03-06 at 11 44 59](https://user-images.githubusercontent.com/231583/223101672-d1bf322f-0375-4607-8690-c8581ed6fd7e.png)